src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs

```diff
@@ -950,14 +962,111 @@ public bool HasSameRules(TimeZoneInfo other)
                 adjustmentRules = (AdjustmentRule[])adjustmentRules.Clone();
             }
 
+            bool hasIanaId = TimeZoneInfo.TryConvertIanaIdToWindowsId(id, out string _);
+
             return new TimeZoneInfo(
                 id,
                 baseUtcOffset,
                 displayName,
                 standardDisplayName,
                 daylightDisplayName,
                 adjustmentRules,
-                disableDaylightSavingTime);
+                disableDaylightSavingTime,
+                hasIanaId);
+        }
+
+        /// <summary>
+        /// Tries to convert IANA time zone Id to Windows Id.
+        /// </summary>
+        /// <param name="ianaId">The IANA time zone Id.</param>
+        /// <param name="windowsId">String object hold the Windows Id which resulted from the IANA Id conversion.</param>
+        /// <returns>True if the Id conversion succeed, false otherwise .</returns>
+        public static unsafe bool TryConvertIanaIdToWindowsId(string ianaId, out string? windowsId)
+        {
+            windowsId = null;
+
+            // This functionality is not enabled in the browser for the sake of size reduction.
+#if !TARGET_BROWSER
```

Comment:
Can we refactor these `#if`s out into separate partial files? We've been trying hard not to use `#if` in these files.

Reply:
This is done